Makefile

```diff
@@ -187,19 +187,36 @@ set-manifest-image:
 ## --------------------------------------
 
 RELEASE_TAG := $(shell git describe --abbrev=0 2>/dev/null)
+RELEASE_GOOS := linux darwin
+RELEASE_DIR := out
+RELEASE_BINARIES := clusterctl
+
+$(RELEASE_DIR):
+	mkdir -p $(RELEASE_DIR)/
 
 .PHONY: release
-release:  ## Builds and push container images using the latest git tag for the commit.
+release: clean-release ## Builds and push container images using the latest git tag for the commit.
 	@if [ -z "${RELEASE_TAG}" ]; then echo "RELEASE_TAG is not set"; exit 1; fi
 	# Push the release image to the staging bucket first.
 	REGISTRY=$(STAGING_REGISTRY) TAG=$(RELEASE_TAG) \
 		$(MAKE) docker-build-all docker-push-all
 	# Set the manifest image to the production bucket.
 	MANIFEST_IMG=$(PROD_REGISTRY)/$(IMAGE_NAME) MANIFEST_TAG=$(RELEASE_TAG) \
 		$(MAKE) set-manifest-image
-	# Generate release artifacts.
-	mkdir out/
-	kustomize build config/default > out/cluster-api-components.yaml
+	$(MAKE) release-manifests
+	$(MAKE) release-binaries
+
+.PHONY: release-manifests
+release-manifests: $(RELEASE_DIR) ## Builds the manifests to publish with a release
+	kustomize build config/default > $(RELEASE_DIR)/cluster-api-components.yaml
+
+.PHONY: release-binaries
+release-binaries: $(addprefix release-binary-,$(RELEASE_BINARIES)) ## Builds the binaries to publish with a release
```

Comment:
Are we expecting multiple binaries here? otherwise these seems like overkill.

Reply:
It's gonna be super easy to port this to other repos in this way, that's why I went down this path